src/gluonts/evaluation/_base.py

```diff
@@ -259,21 +270,22 @@ def get_metrics_per_ts(
         self, time_series: Union[pd.Series, pd.DataFrame], forecast: Forecast
     ) -> Dict[str, Union[float, str, None]]:
         pred_target = np.array(self.extract_pred_target(time_series, forecast))
-        pred_target = np.ma.masked_invalid(pred_target)
-
-        # required for seasonal_error and owa calculation
         past_data = np.array(self.extract_past_data(time_series, forecast))
-        past_data = np.ma.masked_invalid(past_data)
+
+        if self.mask_invalid_values:
+            past_data = np.ma.masked_invalid(past_data)
+            pred_target = np.ma.masked_invalid(pred_target)
 
         try:
-            mean_fcst = forecast.mean
-        except:
+            mean_fcst = getattr(forecast, "mean", None)
+        except NotImplementedError:
```

Comment:
Oh, I see. Now the `getattr` doesn't look so sensible anymore. Maybe we should just not raise an exception for `forecast.mean`.

Reply:
True, although i think that changing the `forecast` goes outside the scope of this pr?

Reply:
Agreed.